plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/AwsEc2UnicastHostsProvider.java

```diff
@@ -74,6 +79,8 @@ public AwsEc2UnicastHostsProvider(Settings settings, TransportService transportS
         this.hostType = HostType.valueOf(settings.get(DISCOVERY_EC2.HOST_TYPE, "private_ip")
                 .toUpperCase(Locale.ROOT));
 
+        this.discoNodes = new DiscoNodesCache(this.settings.getAsTime(NODE_CACHE_TIME_SETTING, TimeValue.timeValueMillis(10_000L)));
```

Comment:
First time I'm seeing `10_000L` instead of `10000L`. Interesting. But can you use `10000L` instead?